File: Ex35ActivityRecyclerView/app/src/main/java/com/hsr2024/ex35activityrecyclerview/MyAdapter.java
package com.hsr2024.ex35activityrecyclerview;

import android.app.Activity;
import android.app.ActivityOptions;
import android.content.Context;
import android.content.Intent;
import android.util.Pair;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.VH> {

    Context context;
    ArrayList<Item> items;

    public MyAdapter(Context context, ArrayList<Item> items) {
        this.context = context;
        this.items = items;
    }

    @NonNull
    @Override
    public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater= LayoutInflater.from(context);
        View view= inflater.inflate(R.layout.recycler_item,parent,false);

        return new VH(view);
    }

    @Override
    public void onBindViewHolder(@NonNull VH holder, int position) {
        Item item= items.get(position);

        holder.iv.setImageResource(item.imgId);
        holder.tvTitle.setText(item.title);
        holder.tvMessage.setText(item.message);

    }

    @Override
    public int getItemCount() {
        return items.size();
    }


    class VH extends RecyclerView.ViewHolder {

        ImageView iv;
        TextView tvTitle,tvMessage;

        public VH(@NonNull View itemView) {
            super(itemView);
            iv= itemView.findViewById(R.id.iv);
            tvTitle= itemView.findViewById(R.id.tv_title);
            tvMessage= itemView.findViewById(R.id.tv_message);

            // 아이템뷰를 클릭하는 것에 반응하기
            itemView.setOnClickListener(v -> {
                //클릭한 아이템뷰의 위치번호를 얻어오기 0,1,2, ....
                int position= this.getLayoutPosition();
                Item item= items.get(position);

                //클릭한 아이템의 정보를 상세하게 보여 줄 화면(Activity)으로 전환
                Intent intent = new Intent(context, DetailActivity.class);
                intent.putExtra("title",item.title);
                intent.putExtra("msg",item.message);
                intent.putExtra("imgId",item.imgId);

                //화면전환할때 전환효과 주기
                ActivityOptions options= ActivityOptions.makeSceneTransitionAnimation((Activity) context,new Pair<View,String>(iv,"ss"));
                context.startActivity(intent, options.toBundle());
                //Toast.makeText(context,position+":"+item.title,Toast.LENGTH_SHORT).show();
            });


        }
    }
}

File: Test60/app/src/main/java/com/hsr2024/test60/MyAdapter.kt
package com.hsr2024.test60

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.hsr2024.test60.databinding.RecyclerViewBinding

class MyAdapter constructor(val context:Context, val itemList:List<Item>): Adapter<MyAdapter.VH>() {

    inner class VH(val binding:RecyclerViewBinding) :ViewHolder(binding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH {
        val binding= RecyclerViewBinding.inflate(LayoutInflater.from(context),parent,false)
        return VH(binding)
    }

    override fun getItemCount(): Int = itemList.size

    override fun onBindViewHolder(holder: VH, position: Int) {
        holder.binding.iv.setImageResource(itemList[position].imgId)
        holder.binding.tv.text = itemList[position].title

        holder.binding.iv.setOnClickListener { Toast.makeText(context,"${itemList[position].title}", Toast.LENGTH_SHORT).show() }

    }
}